src/datasim_ui/views/form.cljs

```diff
@@ -50,6 +51,14 @@
                                           ;; Handle possible auth errors,
                                           ;; or just download the blob result
                                           (condp = (.. e -target -status)
+                                            400 (do
+                                                  (dispatch [:validation/show])
+                                                  (dispatch [:options/hide])
+                                                  (-> (js/Promise.resolve
+                                                       (.text (.. e -target -response)))
+                                                      (.then #(dispatch [:validation/data
+                                                                         (edn/read-string %)])))
+                                                  (snackbar! "Validation Errors Detected, see Validation Results"))
```

Comment:
had to go look at the file... this whole XHR inside an onSubmit handler really needs to get refactored. I know it isn't the focus of this PR to change that but it really should live in a `re-frame.core/reg-fx` handler and get called up from a `re-frame.core/reg-event-fx` handler

Reply:
agreed